### ISSUE TITLE
Number format may be changed inadvertently

### ISSUE THREAD

**Issue by kAlvaro** [phpmyadmin/motranslator#24]
At `\PhpMyAdmin\MoTranslator\Loader::setlocale()` we can find this:

            $this->locale = $locale;
            // Set system locales as well
            if (function_exists('setlocale')) {
                setlocale(0, $locale);
            }

In my Windows box with the official PHP/5.6.21 binaries (php-5.6.21-Win32-VC11-x86) the `LC_...` constant that has zero as value is `LC_ALL`. This basically breaks all code that assumes that numbers gets serialised in locale-independent way, e.g.:

```
/* @var SimpleXMLElement $xml */
/* @var float $amount */
$xml->addChild('amount', $amount); // Implicit cast to string produces: <amount>3,14</amount>
```

`setlocale()` should be called with constants rather than plain integers.

Helper script and sample output:

```
<?php
echo php_uname(), PHP_EOL;
echo PHP_VERSION, PHP_EOL;
$constants = get_defined_constants(true);
asort($constants['standard']);
foreach ($constants['standard'] as $name => $value) {
	if (mb_substr($name, 0, 3) === 'LC_') {
		printf('% -11s = %s', $name, $value); echo PHP_EOL;
	}
}
```

```
Windows NT SOFT2121B 6.2 build 9200 (Unknow Windows version Business Edition) i586
5.3.28
LC_ALL      = 0
LC_COLLATE  = 1
LC_CTYPE    = 2
LC_MONETARY = 3
LC_NUMERIC  = 4
LC_TIME     = 5

Linux SOFT2121b 4.4.0-43-Microsoft #1-Microsoft Wed Dec 31 14:42:53 PST 2014 x86_64
5.5.9-1ubuntu4.21
LC_CTYPE    = 0
LC_NUMERIC  = 1
LC_TIME     = 2
LC_COLLATE  = 3
LC_MONETARY = 4
LC_MESSAGES = 5
LC_ALL      = 6
```

**Comment by kAlvaro** [comment]
Follow-up question... Does the library even need locales at all? This library is particularly useful on Windows when running PHP as Apache module because locales are broken in this architecture (they leak across threads). Fiddling with system locales kind of beats the point.

**Comment by nijel** [comment]
You're right, it should not be part of the library. I've just removed it and released 4.0 with this change.
